### PR TITLE
Call CHECK_NONFATAL only once where needed

### PR DESCRIPTION
**Pull request by maflcko**
Now that `CHECK_NONFATAL` is the identity function starting with commit b1c5991eebb916755be188f355ad36fe01a3f529, it can be called less often in places where it was called more than once on the same value.

### CODE REVIEW

**Review by vincenzopalazzo — APPROVED** [review]
ACK https://github.com/bitcoin/bitcoin/pull/24956/commits/fa3ef7028f8f8d9715999c67da52a90f6fab3e36

**Comment by MarcoFalke** [comment]
Rebased

**Review by jonatack — COMMENTED** [review]
Review ACK fab34d392ca415c27605040dc0fc738016c9a0ca

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -1199,24 +1199,23 @@ RPCHelpMan getblockchaininfo()
...
     obj.pushKV("blocks", height);
     obj.pushKV("headers", chainman.m_best_header ? chainman.m_best_header->nHeight : -1);
-    obj.pushKV("bestblockhash", tip->GetBlockHash().GetHex());
-    obj.pushKV("difficulty", (double)GetDifficulty(tip));
-    obj.pushKV("time", (int64_t)tip->nTime);
-    obj.pushKV("mediantime", (int64_t)tip->GetMedianTimePast());
-    obj.pushKV("verificationprogress", GuessVerificationProgress(Params().TxData(), tip));
+    obj.pushKV("bestblockhash", tip.GetBlockHash().GetHex());
+    obj.pushKV("difficulty", GetDifficulty(&tip));
+    obj.pushKV("time", int64_t{tip.nTime});
```
[the remark this answers is not in the document]
Comment (MarcoFalke) [reply src/rpc/blockchain.cpp:1210]: Thx. I'll leave this for the next force push, if there is one, or a follow-up.

[the remark this answers is not in the document]
Comment (jonatack) [reply src/rpc/blockchain.cpp:1210]: Sure, I'll update #25016 based on this and add it there.

**Pull request merged**